### PR TITLE
feat: port to app-platform

### PR DESCRIPTION
**Pull request by mediremi**
Closes https://jira.dhis2.org/browse/DHIS2-9971

# Changes

## Required due to port

- Updated `AvailableOrganisationUnitsTree` to use `OrganisationUnitTree` from
  `@dhis2/ui` instead of outdated version from `d2-ui` due to dependency issues
- Implemented grid by adding `grid.css`
- Use d2-i18n instead of user settings stored in session storage to get locale
- Removed snackbar and replaced with `useAlert` from `@dhis2/app-runtime`

## Functional changes

- [Fixed endpoint used for unfollowing outliers](https://github.com/dhis2/data-quality-app/pull/702/commits/8f5f61a1041ddec67eafbea0ca8fcaa1245b5063)
- Migrate some buttons, multi select and checkboxes from `material-ui` to `@dhis2/ui`

# Future changes

- Fix handling of back button when sidebar is hidden (sidebar is not restored when going back to form)
- Move responsibility of showing alerts from forms to component responsible for
  API calls (`useAlert` requires functional component as it is a hook but
  converting the various `Page`-extending class components to functional
  components would be out of scope)
- Replace `Sidebar` component from `@dhis2/d2-ui` with custom implementation so
  that `react-router-dom` can be updated
- Replace `@dhis2/d2-ui` and `material-ui` with `@dhis2/ui`
- Get `d2` using `getInstance` instead of React context
- Rename `AlertBar` to make its purpose (notifying user that there are more than 500 results) explicit
- Show spinner when loading outliers

### CODE REVIEW

**Review by Mohammer5 — APPROVED** [review]

File: src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js [hunk 1]
```diff
@@ -1,131 +1,89 @@
...
+                )}
+            </Help>
+        )
     }
 
-    getOrgUnitIdFromPath(path) {
-        return path.split('/').pop()
+    if (data.roots.organisationUnits.length === 0) {
+        return (
+            <p>{i18n.t('You do not have access to any organisation units.')}</p>
```
[the remark this answers is not in the document]
Comment (mediremi) [reply src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js:40]: I moved that error message from [here](https://github.com/dhis2/data-quality-app/pull/702/files/682af8400bc82782f50daf4fd4f5a4b3fc4e92bf#diff-207a3c780a88b33522018da73d75ab90adf1e9b2eb2addb55f61753d30de689bL100-L106) :eyes:

File: src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js [hunk 2]
```diff
@@ -1,131 +1,89 @@
...
             ? selected.filter(selectedPath => selectedPath !== path)
-            : [...this.state.selected, path]
+            : [...selected, path]
+        setSelected(newSelected)
 
-        this.setState({ selected: paths })
-        this.props.onChange &&
-            this.props.onChange(paths.map(this.getOrgUnitIdFromPath))
-    }
+        const getOrgUnitIdFromPath = path => path.split('/').pop()
```
[the remark this answers is not in the document]
Comment (mediremi) [reply src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js:60]: Good catch :+1:

File: src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js [hunk 3]
```diff
@@ -1,131 +1,89 @@
...
-            </div>
-        )
-    }
+AvailableOrganisationUnitsTree.propTypes = {
+    multiselect: PropTypes.bool,
+    onChange: PropTypes.func,
 }
 
-export default AvailableOrganisationUnitsTree
+export default React.memo(AvailableOrganisationUnitsTree)
```
[the remark this answers is not in the document]
Comment (mediremi) [reply src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js]: `AvailableOrganisationUnitsTree` was a `PureComponent` so when I converted it to a functional component I added `React.memo`. Removing it doesn't impact performance so I've removed it.

File: src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mohammer5) [reply src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js:40]: :+1:

**Pull request merged**